covasim/sim.py

```diff
@@ -52,7 +52,16 @@ def __init__(self, pars=None, datafile=None, datacols=None, popfile=None, filena
         self.load_data(datafile, datacols) # Load the data, if provided
         self.load_population(popfile)      # Load the population, if provided
         self.update_pars(pars)             # Update the parameters, if provided
-        return
+
+        # if resampling is enabled, make 100k(default) pop size
+        if self["resample"] and self["pop_size"] > self["resample_max_pop"]:
+
+            self.max_n = self["pop_size"]
+            self["pop_size"] = self["resample_max_pop"]
+
+        # dynamic resampling
+        self.resampe_threshold = self["resample_threshold"]  # Fraction of total pop that isn't susceptible when we trigger scaling
+        self.resample_factor = self["resample_factor"]  # Factor by which we will rescale population
```

Comment:
Not sure why these are being pulled out and set as attributes instead of leaving them in the dict?